org.adempiere.base/src/org/compiere/model/MCost.java

```diff
@@ -150,6 +203,13 @@ protected static BigDecimal getCurrentCost (MProduct product, int M_ASI_ID,
 		BigDecimal otherCostEach = Env.ZERO;
 		BigDecimal percentage = Env.ZERO;
 		int count = 0;
+		
+		// get cost history record based on the account date
+		if (history == null && dateAcct != null) {
+			history = getCostHistory(product.getCtx(), product.getAD_Client_ID(), Org_ID, product.getM_Product_ID(),
+					M_CostType_ID, as.getC_AcctSchema_ID(), costingMethod, M_ASI_ID, dateAcct, trxName);
+		}
+		

```

Comment:
If only dateAcct parameter is needed, it would make sense to remove the history parameter. Would simplify usage of the method and reduce duplication of code.

Reply:
The passed in cost history record is retrieved based on the cost detail ID. If cost history parameter is NULL, we get the cost history record based on the account date.